react-game/react-game/src/Components/AppContent.js

```diff
@@ -0,0 +1,136 @@
+import React, {useState} from 'react';
+import {Cards} from './Cards';
+import Score from './Score';
+import ScoreList from './ScoreList';
+import enterFullscreen from '../fullscreen';
+import Settings from './SettingBtn';
+
+
+
+function ModalEndGame({gameover, score, pushscore}) {
+
+  const [value, setValue] = useState('');
+
+ function onchange(e){
+    setValue(e.target.value);
+  }
+
+  return (
+   <div className={gameover ? "modal-game active" : "modal-game "}>
+    <div className={gameover ? "modal-game__content active" : "modal-game __content"} >
+    <h2>Поздравляем, вы набрали {score} очков</h2> 
+    <p>Введите ваше Имя</p>
+    <form onSubmit={(event)=> event.preventDefault()}>
+    <input className='inputscoreN' type="text" placeholder='Как вас зовут?' onChange={onchange}/><button className='btn-submitscore' type="submit" onClick={() => pushscore(value)}>Ok!</button>
+      </form>
+    </div>
+ </div>
+  )
+ }
+
+export default class AppContent extends React.Component {
+
+ constructor(props) {
+  super(props);
+  this.state = JSON.parse(localStorage.getItem('state')) ? JSON.parse(localStorage.getItem('state')) : {
+    cards: [21,22,23,24,31,32,33,34,41,42,43,44,51,52,53,54,61,62,63,64,71,72,73,74,81,82,83,84,91,92,93,94,101,102,103,104,'j1', 'j2', 'j3', 'j4', 'q1','q2','q3','q4','k1','k2','k3','k4','a1','a2','a3','a4']
+  ,score: 0, timer: 300, stoptimer: false, newgame: false, gameover: false, scoreList:[], volume:40};
+  this.timers = this.state.timer < 300 ? setInterval(()=> {
+    this.setState({timer: this.state.timer - 1});
+    localStorage.setItem('state',JSON.stringify(this.state));
+  if (this.state.timer === 0 || this.state.stoptimer) {
+    clearInterval(this.timers)}
+    if (this.state.timer === 0) {this.gameOver.call(this);}
+  }, 1000) : null;
+  this.setState({cards: this.state.cards.sort(() => Math.random() - 0.5)});
+}
+
+deletecard() {
+  let arr = this.state.cards;
+  let a = arr.pop();
+  this.setState({cards: arr});
+  localStorage.setItem('state',JSON.stringify(this.state));
+  if (this.state.cards.length === 0) {this.gameOver.call(this);}
+ return a;
+}
+
+sumScore(e) {
+this.setState({score: this.state.score += e});
+localStorage.setItem('state',JSON.stringify(this.state));
+}
+
+stoptimer() {
+  this.setState({stoptimer: true});
+  localStorage.setItem('state',JSON.stringify(this.state));
+  return this.state.timer;

```

Comment:
this function should stop timer, we do not need to return this.state.timer to pass it via props in Card component (131 row). It is better to pass this this.state.timer directly in props. More over setStae is asynchronous operation, there is no guarantee that on the moment of returning you state is already updated.